app/jobs/comments/comment_job.rb

```diff
@@ -0,0 +1,14 @@
+module Comments
+  class CommentJob < ApplicationJob
+    queue_as :default
+
+    def self.perform_later(*args)
+      super unless Rails.env.development?
+    end
+
+    def perform(comment_id, method, *args)
+      comment = Comment.find_by(id: comment_id)
+      comment&.public_send(method.to_sym, *args)
```

Comment:
Mmm I think this is a neat trick but it's a bit like hiding the dust under the carpet if you know what I mean :D 
This is my line of reasoning:

Now we have a bunch of methods that are magically made async by DelayedJob, the goals for what I've understood all the work @lightalloy is doing are:

* make the code more explicit
* create single responsibility jobs
* have queues for each task 

This way we're just replacing `handle_asynchronously` of DJ without actually getting at the bulk of the issue. 

I think there shouldn't be a generic "CommentJob" that can basically trigger any public method inside the class, we should instead make it so that each async task is handled separately.

Otherwise we could just have a generic "AnythingJob" that gets passed a class, a method name and some arguments and calls anything with async, which is basically what DJ already does that they are trying to move away from eheh.

My suggestion though is to split this class in multiple jobs and have each job call a service. You can get inspiration and guidance on how to do it by reading these two comments: https://github.com/thepracticaldev/dev.to/issues/2497#issuecomment-485396688 https://github.com/thepracticaldev/dev.to/issues/2497#issuecomment-485398507